crates/rpc/rpc/src/eth/signer.rs

```diff
@@ -26,14 +32,35 @@ pub(crate) trait EthSigner: Send + Sync {
         request: TypedTransactionRequest,
         address: &Address,
     ) -> Result<TransactionSigned>;
+
+    /// Encodes and signs the typed data according EIP-712. Payload must implement Eip712 trait.
+    fn sign_typed_data(&self, address: Address, payload: &TypedData) -> Result<Signature>;
 }
 
 /// Holds developer keys
 pub(crate) struct DevSigner {
     addresses: Vec<Address>,
     accounts: HashMap<Address, SecretKey>,
+    sign_engine: Secp256k1<All>,
 }
 
+impl DevSigner {
```

Comment:
Does the signer need to own the chain id to apply EIP-155? Maybe it's something we do in a follow-up?

Reply:
perhaps a reasonable assumption here is that the Signer should do this.

Reply:
This would be needed only for `eth_signTransaction` right?

Reply:
yes.